### ISSUE TITLE
Cypress cucumber test hanging in between run. when test running headed mode.

### ISSUE THREAD

**Issue by VaghasiyaKrupal** [cypress-io/cypress#21353]
### Current behavior

I have implemented the cucumber framework in my cypress project. But the issue is that when I run some script it will hang in the test runner when the test is running in headed mode.

I don't know what is the problem here. If anyone has a solution for that please suggest me. When I run all `.feature` files together it will crash the test runner and show like the below image.

![image](https://user-images.githubusercontent.com/45596703/166976238-c2c7cb71-bec7-4ad6-9f75-e6aa9350c62b.png)

### Test code to reproduce

To reproduce the issue please follow the below steps.
1. clone `https://github.com/VaghasiyaKrupal/cucumber-issue.git`
2. Install all dependacy using `npm install`
3. run `npx cypress open`
4. Run `Builder.feature` OR `ProjectProfile.feature` OR `Event.feature` file

### Cypress Version

9.5.4

**Comment by davidmunechika** [comment]
I am not able to reproduce this exact issue locally. At first it appeared your tests were hanging, but I think they are just running really slowly due to the timeouts that are set- you have your `defaultCommandTimeout` set to 2 minutes and your `requestTimeout` and `responseTimeout` set to 5 minutes. For certain commands like `cy.react()` which will ultimately use the entire time to look for a component that does not exist, this will likely make it seem like your tests are hanging. I was able to get all tests to finish execution if I waited long enough.

As for the issue regarding crashing the test runner, I was not able to reproduce using the steps you provided. Is there any more information you could provide for this specific problem?

**Comment by VaghasiyaKrupal** [comment]
@davidmunechika issue is not related to timeout. This is maximum timeout i have given.

**Comment by davidmunechika** [comment]
Yes, I ran all of the spec files together as well as the three you listed in the description in isolation. I was not able to reproduce a situation where the test runner crashes.

**Comment by VaghasiyaKrupal** [comment]
@davidmunechika I am facing the same issue every time. but every time test runner does not crash. Sometimes it will show tests are running but actually, it will be hanged and process infinity. so I have to close the test runner and re-run the test.

**Comment by BlueWinds** [comment]
tl;dr: `cypress-react` is the culprit, a 3rd party library that we don't directly support. You'll have to ask them.

Longer response:

I've been able to reproduce myself using the provide repo. The issue appears to be with the (cypress-react-selector)[https://www.npmjs.com/package/cypress-react-selector] plugin, rather than with cypress itself. For example, with the scenario

```
  Scenario: Empty builder's answer [Builder'sAnswer]
    Given Visit builder answer page
    When Close the notification
    And Prop project ID
    Then Find paragraph from answer
```

if I remove the `And Prop project ID` line - which resolves to `cy.react('BuildersQuillForm', { props: { projectId: 1511 } });` - then the test passes. I'm not familiar with this plugin, but you'll likely have more success reaching out to the maintainers of that project.

A couple of other comments.
1. The `cypress-xpath` appears to be quite slow. Your tests will run much, much faster if you replace this with the builtin cypress commands. You can chain commands together to achieve similar effects without a plugin: `cy.get('.article').find('footer')`, rather than complicated xpath strings.
2. In general it's far perferable to interact with the DOM rather than with your component structure. Even if it were working without issue, we'd still recommend the use of (Cypress Testing Library)[https://testing-library.com/docs/cypress-testing-library/intro/] over cypress-react.

Hopefully some of this is helpful!

**Comment by VaghasiyaKrupal** [comment]
@BlueWinds I am using `cy.react` in only builder script but also problem for other script where I have not use `cy.react`.

**Comment by VaghasiyaKrupal** [comment]
@Please recheck i am facing same issue every time
![image](https://user-images.githubusercontent.com/45596703/168316053-f4488237-ba74-44e6-b313-63fd2a219c1b.png)
